uitools/toolkitcpp.cmake

```diff
@@ -0,0 +1,118 @@
+    ${CPPPATH}/NorthArrowController.cpp
+    ${CPPPATH}/OverviewMapController.cpp
+    ${CPPPATH}/PopupViewController.cpp
+    ${CPPPATH}/ScalebarController.cpp
+    ${CPPPATH}/SearchResult.cpp
+    ${CPPPATH}/SearchSourceInterface.cpp
+    ${CPPPATH}/SearchSuggestion.cpp
+    ${CPPPATH}/SearchViewController.cpp
+    ${CPPPATH}/SmartLocatorSearchSource.cpp
+    ${CPPPATH}/TimeSliderController.cpp
+    ${CPPPATH}/UtilityNetworkFunctionTraceResult.cpp
+    ${CPPPATH}/UtilityNetworkFunctionTraceResultsModel.cpp
+    ${CPPPATH}/UtilityNetworkListItem.cpp
+    ${CPPPATH}/UtilityNetworkTraceController.cpp
+    ${CPPPATH}/UtilityNetworkTraceStartingPoint.cpp
+    ${CPPPATH}/UtilityNetworkTraceStartingPointsModel.cpp
+    ${REGISTERPATH}/register.cpp
+    ${REGISTERPATH}/internal/register_cpp.cpp)
+
+set(TOOLKITCPP_HEADERS
+    ${CPPPATH}/AuthenticationController.h
+    ${CPPPATH}/BasemapGalleryController.h
+    ${CPPPATH}/BasemapGalleryItem.h
+    ${CPPPATH}/BookmarksViewController.h
+    ${CPPPATH}/BookmarkListItem.h
+    ${CPPPATH}/CoordinateConversionConstants.h
+    ${CPPPATH}/CoordinateConversionController.h
+    ${CPPPATH}/CoordinateConversionOption.h
+    ${CPPPATH}/CoordinateConversionResult.h
+    ${CPPPATH}/CoordinateOptionDefaults.h
+    ${CPPPATH}/FloorFilterController.h
+    ${CPPPATH}/FloorFilterFacilityItem.h
+    ${CPPPATH}/FloorFilterLevelItem.h
+    ${CPPPATH}/FloorFilterSiteItem.h
+    ${CPPPATH}/LocatorSearchSource.h
+    ${CPPPATH}/Internal/BasemapGalleryImageProvider.h
+    ${CPPPATH}/Internal/DisconnectOnSignal.h
+    ${CPPPATH}/Internal/DoOnLoad.h
+    ${CPPPATH}/Internal/GenericListModel.h
+    ${CPPPATH}/Internal/GenericTableProxyModel.h
+    ${CPPPATH}/Internal/GeoViews.h
+    ${CPPPATH}/Internal/MetaElement.h
+    ${CPPPATH}/Internal/SingleShotConnection.h
+    ${CPPPATH}/NorthArrowController.h
+    ${CPPPATH}/OverviewMapController.h
+    ${CPPPATH}/PopupViewController.h
+    ${CPPPATH}/ScalebarController.h
+    ${CPPPATH}/SearchResult.h
+    ${CPPPATH}/SearchSourceInterface.h
+    ${CPPPATH}/SearchSuggestion.h
+    ${CPPPATH}/SearchViewController.h
+    ${CPPPATH}/SmartLocatorSearchSource.h
+    ${CPPPATH}/TimeSliderController.h
+    ${CPPPATH}/UtilityNetworkFunctionTraceResult.h
+    ${CPPPATH}/UtilityNetworkFunctionTraceResultsModel.h
+    ${CPPPATH}/UtilityNetworkListItem.h
+    ${CPPPATH}/UtilityNetworkTraceController.h
+    ${CPPPATH}/UtilityNetworkTraceStartingPoint.h
+    ${CPPPATH}/UtilityNetworkTraceStartingPointsModel.h
+    ${REGISTERPATH}/register.h
+    ${REGISTERPATH}/internal/register_cpp.h)
+
+set(TOOLKITCPP_RESOURCES
+    ${CMAKE_CURRENT_LIST_DIR}/images/esri_arcgisruntime_toolkit_images.qrc
+    ${CMAKE_CURRENT_LIST_DIR}/import/Esri/ArcGISRuntime/Toolkit/esri_arcgisruntime_toolkit_view.qrc)
+
+target_sources(${PROJECT_NAME} PRIVATE
+    ${TOOLKITCPP_SOURCES}
+    ${TOOLKITCPP_HEADERS}
+    ${TOOLKITCPP_RESOURCES})
+
+find_package(Qt6 COMPONENTS REQUIRED quickcontrols2 webview svg quicklayouts)
```

Comment:
This relies on the `include()` call for the outer `CMakeLists.txt` file being after we call `qt_add_executable()`. We can either try and make this more resilient against where the user puts this `include()`, or we could update our templates to have a comment showing them where to put the `include()`

Reply:
I think ideally the solution we come up with would just be able to be included anywhere.

Reply:
I agree it should be able to be inserted anywhere, within the parameters of idiomatic CMake. I'm not strong with CMake, so there may be normal CMake restrictions about this that are well-known to CMake folks that we don't need to worry about.

Reply:
Looking at it again, it might be best to have this available through a package. I had experimented with that before going with `include()`, but had hit a snag somewhere. But it seems like it would make more sense to do something like this
```
find_package(ArcGISRuntime 200.5.0 COMPONENTS REQUIRED Cpp)
find_package(ArcGISToolkit 1.0.0 something something)
```

rather than this
```
find_package(ArcGISRuntime 200.5.0 COMPONENTS REQUIRED Cpp)
include(<path to cmake>)
```

and then the placement of the `find_package` seems to come naturally.